components/omega/external/CMakeLists.txt

```diff
@@ -12,19 +14,11 @@ add_subdirectory(
   ${CMAKE_CURRENT_BINARY_DIR}/ekat/extern/yaml-cpp
 )
 
-# Add the YAKL library
-add_subdirectory(
-  ${E3SM_EXTERNALS_ROOT}/YAKL
-  ${CMAKE_CURRENT_BINARY_DIR}/YAKL
-)
-
-if (CMAKE_VERSION VERSION_GREATER "3.18.0")
-  set_property(TARGET yakl PROPERTY CUDA_ARCHITECTURES OFF)
-endif()
-
-if(OMEGA_DEBUG)
-  target_compile_definitions(yakl INTERFACE YAKL_DEBUG=1)
-endif()
+# Add the Kokkos library
```

Comment:
Are all these tpls explicitly handled only in a standalone build or is this code executed regardless? If the latter, then you should add some cmake logic that prevents the tpls to be processed twice (once in Omega and once in eamxx). Someone like 

```cmake
if (NOT TARGET Kokkos::kokkos)
  # handle kokkos
endif()
```

Or you can use Ekat's cmake scripts (you can look at cake/tpls/ElatTPLs.cmake in the ekat repo, in case this is relevant)

Reply:
The part is executed for both standalone and E3SM builds. I agree that we need a way to prevent the TPLs from being processed twice. I will look at the Ekat script. Thank you very much for sharing your insight on this issue.

Reply:
I have updated the Omega TPL builds with your suggested if-blocks. I think I may need to check whether using the Ekat script could be a better solution when Omega is merged into E3SM.